packages/terra-form/src/Select.jsx

```diff
@@ -62,7 +62,7 @@ const Select = ({
       defaultValue={defaultValue}
       className={classNames('terra-Form-select', additionalSelectFieldProps.className)}
     >
-      {choices.map(choice => <option key={choice.toString()} value={choice}>{choice}</option>)}
+      {choices.map(choice => <option key={choice.value} value={choice.value}>{choice.display}</option>)}
```

Comment:
Ok yep I should definitely have a value prop for controlled forms. They should work like all the other inputs now, pass an onchange and value prop.